src/sentry/static/sentry/app/views/eventsV2/eventDetails/transaction/eventBreakdown.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+import get from 'lodash/get';
+
+import {Event} from 'app/types';
+import {
+  SentryTransactionEvent,
+  SpanEntry,
+  SpanType,
+} from 'app/components/events/interfaces/spans/types';
+import {TraceContextType} from 'app/components/events/interfaces/spans/traceView';
+
+type EventBreakdownType = {
+  ops: {name: string; count: number; percentage: number; duration: number}[];
+  unknown: {count: number; percentage: number; duration: number} | undefined;
+};
+
+type Props = {
+  event: Event;
+};
+
+class EventBreakdown extends React.Component<Props> {
+  getTransactionEvent(): SentryTransactionEvent | undefined {
+    const {event} = this.props;
+
+    if (event.type === 'transaction') {
+      return event as SentryTransactionEvent;
+    }
+
+    return undefined;
+  }
+
+  generateCounts(): EventBreakdownType {
+    const event = this.getTransactionEvent();
+
+    if (!event) {
+      return {
+        ops: [],
+        unknown: undefined,
+      };
+    }
+
+    const spanEntry: SpanEntry | undefined = event.entries.find(
+      (entry: {type: string}) => entry.type === 'spans'
+    );
+
+    const spans: SpanType[] = get(spanEntry, 'data', []);
+
+    // track stats on spans with no operation
+    const spansWithNoOperation = {count: 0, duration: 0};
+
+    const numOfSpans = spans.length;
+
+    type AggregateType = {
```

Comment:
You name this type here, but couldn't it also be used in the EventBreakdown type?

Reply:
`EventBreakdown` is array based whereas `AggregateType` is mapped based.